Updates GUI installation and configuration documents (#1442)
dotse/zonemaster: docs/public/configuration/gui/configuring-using-config-ts.md

# Simple build-time configuring Zonemaster-GUI using "config.ts"

The `config.ts` configuration file can be used for *build-time* settings of the

> Zonemaster-GUI, i.e. the installation package has to be rebuilt.

Reviewer: ```suggestion
The `config.ts` configuration file can be used for *build-time* settings of
the Zonemaster-GUI. Any changes to this file require a rebuild of the
installation package.
```

## Finding "config.ts"

In the source tree the file is found as `config.ts` relative to the root of the
Git tree. It does not exist in the installed GUI, as the values have been built

> in

Reviewer: ```suggestion
In the source tree the file is found as `config.ts` relative to the root of
the Git tree. The installed GUI does not ship with this file, because the
values defined in this file are baked in the GUI’s HTML and JavaScript files.
```

## Rebuilding Zonemaster-GUI after update

> When `config.ts` has been updated, Zonemaster-GUI has to be rebuilt. See

Reviewer: ```suggestion
When `config.ts` is updated, Zonemaster-GUI has to be rebuilt. See
```

  * The default language must be one of the enabled languages.
  * The default value is "en".
  * If this option is updated see [Updating Apache configuration] for required
    matching update of the Apache configuration.
* **enabledLanguages**: An array of language codes that are supported by the
  application.
  * The array should match the list of language codes in the [Locale setting] in
    the Backend configuration.
  * The array must only include language codes also included in
    `project.inlang/settings.json`, but it may be fewer. To add new languages,
    see [Translating Zonemaster-GUI]. `project.inlang/settings.json` must only be
    updated in that process.
* **baseUrl**: The base path Zonemaster-GUI is served on. By default it is "/".
  * If Zonemaster-GUI should be served on e.g. `http:/domaintest.xa/zonemaster`
    instead of just `http://domaintest.xa/` then this option to be set to
    "/zonemaster".
  * If this option is updated see [Updating Apache configuration] for required
    matching update of the Apache configuration.
* **apiBaseUrl**: The base URL for API requests. The defaults value is '/api'

>   and must be kept so if `baseUrl` has its default value.

Reviewer: ```suggestion
* **apiBaseUrl**: The base URL for API requests. The default value is `/api`
  and must be kept so if `baseUrl` has its default value.
```

        siteName: '',
    },
};
```

## Configuration Options

* **defaultLanguage**: The default language to use when no language is specified.
  * The default language must be one of the enabled languages.
* **enabledLanguages**: An array of language codes that are supported by the
  application.
  * The array should match the list of language codesin the [Locale setting] in
    the Backend configuration.
  * The array must only include language codes also included in
    `project.inlang/settings.json`, but it may be fewer. To add new languages,
    see the [Translating GUI]. `project.inlang/settings.json` must only be
    updated in that process.
* **apiBaseUrl**: The base URL for API requests. This is taken from the
  `PUBLIC_API_URL` environment variable (default empty) or defaults to '/api'.
  * If this value is changed, then [zonemaster.conf-example] must also be
  updated. More information below.
* **pollingInterval**: The interval (in milliseconds) for polling the API. This
  is taken from the `PUBLIC_POLLING_INTERVAL` environment variable (default
  empty) or defaults to 5000 ms.
* **clientInfo**: Information about the client application. Both are best left

>   as default.

Reviewer 2: Maybe say something about how this is used?
```suggestion
* **clientInfo**: Information about the client application. These details are
  included in every API request to start tests as (spoofable) indications of
  origin. Both are best left as default.
```

    `project.inlang/settings.json`, but it may be fewer. To add new languages,
    see [Translating Zonemaster-GUI]. `project.inlang/settings.json` must only be
    updated in that process.
* **apiBaseUrl**: The base URL for API requests. This is taken from the
  `PUBLIC_API_URL` environment variable (default empty) or defaults to '/api'.
  * If this value is changed, then [zonemaster.conf-example] must also be
  updated. More information below.
* **pollingInterval**: The interval (in milliseconds) for polling the API. This
  is taken from the `PUBLIC_POLLING_INTERVAL` environment variable (default
  empty) or defaults to 5000 ms.
* **clientInfo**: Information about the client application. These details are
  included in every API request to start tests as (spoofable) indications of
  origin. Both are best left as default.
  * **version**: The version of the application, taken from package.json.
  * **id**: The identifier for the client application.
* **siteInfo**: Information about the site.
  * **email**: The contact email address. To be set in the footer in a mailto
    URL, if non-empty.
  - **siteName**: The name of the site. To be set in the header if non-empty.

> ### Setting "apiBaseUrl"

Author: Could be either-or.
